Stack/Opc.Ua.Core/Security/Certificates/CertificateFactory.cs

```diff
@@ -426,10 +420,13 @@ string password
                 // merge first cert with private key into X509Certificate2
                 certificate = new X509Certificate2(
                     rawData,
-                    (password == null) ? String.Empty : password,
+                    password ?? String.Empty,
                     storageFlags[flagsRetryCounter]);
                 // can we really access the private key?
-                using (RSA rsa = certificate.GetRSAPrivateKey()) { }
+                if (VerifyRSAKeyPair(certificate, certificate, true))
+                {
+                    return certificate;
+                }
```

Comment:
if not, throw...

Reply:
it throws, its ok

---
In reply to: [155167096](https://github.com/OPCFoundation/UA-.NETStandard/pull/284#discussion_r155167096) [](ancestors = 155167096)